alchemist/alchemist-swingui/src/main/java/it/unibo/alchemist/boundary/gui/isolines/conrec/Conrec.java

```diff
@@ -0,0 +1,269 @@
+/*
+ * Copyright (C) 2010-2019, Danilo Pianini and contributors
+ * listed in the main project's alchemist/build.gradle.kts file.
+ *
+ * This file is part of Alchemist, and is distributed under the terms of the
+ * GNU General Public License, with a linking exception,
+ * as described in the file LICENSE in the Alchemist distribution's top directory.
+ */
+
+package it.unibo.alchemist.boundary.gui.isolines.conrec;
+
+import java.util.Objects;
+
+/**
+ * Conrec a straightforward method of contouring some surface represented a regular
+ * triangular mesh.
+ *
+ * Ported from the C++ code by Nicholas Yue (see above copyright notice).
+ * See <a href="http://paulbourke.net/papers/conrec">http://paulbourke.net/papers/conrec</a> for full description
+ * of code and original C++ source.
+ *
+ * author  Bradley White
+ * @version 1.0
+ */
+public class Conrec {
+
+    private static final int ARR_SIZE = 5;
+    private static final int CASE_1 = 1;
+    private static final int CASE_2 = 2;
+    private static final int CASE_3 = 3;
+    private static final int CASE_4 = 4;
+    private static final int CASE_5 = 5;
+    private static final int CASE_6 = 6;
+    private static final int CASE_7 = 7;
+    private static final int CASE_8 = 8;
+    private static final int CASE_9 = 9;
+    private static final int[][][] CASTAB =
```

Comment:
WTF is this?

Reply:
You mean `CASTAB`? I have no idea, I think they are using it to determine which `CASE` they are falling in, so the numbers stored in that array are basically different cases, but how the whole thing works is obscure. Should I shed light on it?

Reply:
Obscure code inside something we maintain is super bad.
What we should do is the following:

1. check if a library doing what we need exists, if so, use it
2. otherwise, check the license of this source, if possible, make a library with it
3. use the library as a black box